packages/react-ui-core/src/Ratings/RatingBar.js

```diff
@@ -23,14 +26,24 @@ export default class RatingBar extends PureComponent {
     score: 0,
     maxScore: 5,
     RatingItem: Star,
+    fillColor: 'black',
+  }
+
+  constructor(props) {
+    super(props)
+    this.effectiveId = this.props.id || randomId()
+  }
```

Comment:
@tadjohnston, i'm having lots of trouble making this generated id work properly.  Here's what i see happening:  on SSR, i generate a new id, it gets passed down to `Star` where it is properly set on the `LinearGradient` and the reference is correct in the svg's `path`.  The, on the client side the constructor runs again, a new id gets generated and `LinearGradient` gets updated with it.  However, the svg's `path` does not re-render with the new id, it still has the old id.  That breaks things.   Also, isn't this causing a mismatch between client/server rendering?   If i pass in an id rather than generating one, everything works properly.